File: 1170.比较字符串最小字母出现频次.py
#
# @lc app=leetcode.cn id=1170 lang=python3
#
# [1170] 比较字符串最小字母出现频次
#

# @lc code=start
class Solution:
    def count_characters(self, s: str) -> int:
        counter=Counter(s)
        min_char=min(counter.keys())
        return counter[min_char]
    def numSmallerByFrequency(self, queries: List[str], words: List[str]) -> List[int]:
        words_cnt=[]
        for s in words:
            words_cnt.append(self.count_characters(s))

        n=len(words_cnt)
        words_cnt.sort()
        ans=[]
        for s in queries:
            idx=bisect_right(words_cnt,self.count_characters(s))
            ans.append(n-idx)

        return ans
# @lc code=end

File: 1170.比较字符串最小字母出现频次.cpp
/*
 * @lc app=leetcode.cn id=1170 lang=cpp
 *
 * [1170] 比较字符串最小字母出现频次
 */

// @lc code=start
    int count_characters(string s) {
        map<char, int> counter;
        for (char c: s) {
            counter[c]++;
        }
        return counter.begin()->second;
    }
class Solution {
public:
    vector<int> numSmallerByFrequency(vector<string>& queries, vector<string>& words) {
        vector<int> words_cnt(words.size(),0),ans;
        for (string s: words) {
            words_cnt.push_back(count_characters(s));
        }
        sort(words_cnt.begin(), words_cnt.end());
        int n = words_cnt.size();
        for (string s : queries) {
            auto idx = upper_bound(words_cnt.begin(),words_cnt.end(),count_characters(s));
            ans.push_back(n - (idx-words_cnt.begin()));
        }
        return ans;
    }
};
// @lc code=end

